equinix/resource_fabric_connection_acc_test.go

```diff
@@ -0,0 +1,281 @@
+package equinix
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"testing"
+	"time"
+
+	v4 "github.com/equinix-labs/fabric-go/fabric/v4"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccFabricCreateConnection(t *testing.T) {
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: checkConnectionDelete,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccFabricCreateAzureConnectionConfig(50),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"equinix_fabric_connection.test", "name", fmt.Sprint("fabric_tf_acc_Generic")),
+					resource.TestCheckResourceAttr(
+						"equinix_fabric_connection.test", "bandwidth", fmt.Sprint("50")),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+			{
+				Config: testAccFabricCreateAzureConnectionConfig(100),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"equinix_fabric_connection.test", "name", fmt.Sprint("fabric_tf_acc_Generic")),
+					resource.TestCheckResourceAttr(
+						"equinix_fabric_connection.test", "bandwidth", fmt.Sprint("100")),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func checkConnectionDelete(s *terraform.State) error {
+	client := testAccProvider.Meta().(*Config).fabricClient
+	ctx := context.Background()
+	ctx = context.WithValue(ctx, v4.ContextAccessToken, testAccProvider.Meta().(*Config).FabricAuthToken)
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "equinix_fabric_connection" {
+			continue
+		}
+		_, err := waitUntilConnectionDeprovisioned(rs.Primary.ID, client, ctx)
```

Comment:
This wait function is being tested but not implemented in the process of deleting the connection resource